### ISSUE TITLE
[Bug][iOS][A11Y][Phone Number] Keyboard focus reset

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-flutter#499]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

iPhone 14+

### OS Version

iOS26

### App version

0.6.0 (509)

### Describe the issue

2 main issues on phone number using an external keyboard (after activating Full Keyboard Access) :

- When I open the country selector, the focus is reset and is put on the main view
- When I change country, the focus is reset too
- Once I entered phone number, I press enter to leave the edition mode, the focus is reset too

### Expected Behavior

- when I open the country selector: the focus should be put on the current selected country
- when I close the country selector, the focus should stay on the country selector
- when I press enter to leave the edition mode, the focus shall stay on input field

**Comment by AhmedAmineZr** [comment]
@pya35 I make a resolve do the first comment of Benoit, it can resolve this issue  https://github.com/Orange-OpenSource/ouds-flutter/issues/326#issuecomment-3571663811
